yfinance/base.py

```diff
@@ -916,7 +927,7 @@ def get_earnings_dates(self, proxy=None) -> Optional[pd.DataFrame]:
 
         dates = dates.set_index("Earnings Date")
 
-        self._earnings_dates = dates
+        self._earnings_dates[limit] = dates
```

Comment:
I don't understand purpose of this 'dict cache'. data.py is already caching the data.

Reply:
Caching this here will avoid doing the reparsing of the html table and the data manipulation.
Also the data cache is limited to the 64 latest unique requests so if you have some ticker instances active it will result in cache misses.